src/day13.scala

```diff
@@ -0,0 +1,66 @@
+// using scala 3.0.2
+
+package day13
+
+import scala.util.Using
+import scala.io.Source
+
+@main def part1(): Unit =
+  val answer = part1(readInput())
+  println(s"The answer is: $answer")
+
+@main def part2(): Unit =
+  val answer = part2(readInput())
+  println(s"The answer is:\n$answer")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day13"))(_.mkString)
+
+def part1(input: String): Int =
+  val (dots, folds) = parseInstructions(input)
+  dots.map(folds(0).apply).size
+
+def part2(input: String): String =
+  val (dots, folds) = parseInstructions(input)
+  val foldedDots = folds.foldLeft(dots)((dots, fold) => dots.map(fold.apply))
+  
+  val (length, height) = (foldedDots.map(_.x).max + 1, foldedDots.map(_.y).max + 1)
+  val paper = Array.fill(height, length)('.')
+  for dot <- foldedDots do paper(dot.y)(dot.x) = '#'
+  
+  paper.map(_.mkString).mkString("\n")
+
+def parseInstructions(input: String): (Set[Dot], List[Fold]) =
+  val sections = input.split("\n\n")
+  val dots = sections(0).split('\n').toSet.map(Dot.parse)
+  val folds = sections(1).split('\n').toList.map(Fold.parse)
+  (dots, folds)
+
+def fold(along: Int)(point: Int): Int =
+  if point < along then point
+  else 2 * along - point
```

Comment:
My intuition here would be to write `along - (point - along)`. Of course these are equivalent, but I can only convince myself that `2 * along - point` is correct by verifying that it is equivalent to `along - (point - along)`. Perhaps add a comment or something?